L1Trigger/L1TMuonEndCap/src/SingleHitTrack.cc

```diff
@@ -32,120 +32,124 @@ void SingleHitTrack::process(
+      if (already_used)
+	continue;
+      
+      int zone = -1;
+      int zone_code = conv_hits_it->Zone_code();
+      if      (zone_code & 0b1000) zone = 4;
+      else if (zone_code & 0b0100) zone = 3;
+      else if (zone_code & 0b0010) zone = 2;
+      else if (zone_code & 0b0001) zone = 1;
+      else {
+	std::cout << "\n\n EMTF SingleHitTrack.cc - bizzare case where zone_code = " << zone_code << std::endl;
+	assert(zone > 0);
       }
-    } // End loop: for (; best_tracks_it != best_tracks_end; ++best_tracks_it)
-
-    // Only use hits that have not been used in a track
-    if (already_used)
-      continue;
-
-    int zone = -1;
-    int zone_code = conv_hits_it->Zone_code();
-    if      (zone_code & 0b1000) zone = 4;
-    else if (zone_code & 0b0100) zone = 3;
-    else if (zone_code & 0b0010) zone = 2;
-    else if (zone_code & 0b0001) zone = 1;
-    else {
-      std::cout << "\n\n EMTF SingleHitTrack.cc - bizzare case where zone_code = " << zone_code << std::endl;
-      assert(zone > 0);
-    }
-
-    // Set "mode" using CLCT bend
-    int mode = -1;
-    int CLCT = conv_hits_it->Pattern();
-    if      (CLCT >= 8) mode = 8;
-    else if (CLCT >= 6) mode = 4;
-    else if (CLCT >= 4) mode = 2;
-    else if (CLCT >= 2) mode = 1;
-    else {
-      std::cout << "\n\n EMTF SingleHitTrack.cc - bizzare case where CLCT = " << CLCT << std::endl;
-      assert(mode > 0);
-    }
-
-
-    EMTFTrack new_trk;
-    new_trk.push_Hit ( *conv_hits_it );
-
-    EMTFPtLUT empty_LUT{};
-    new_trk.set_PtLUT ( empty_LUT );
-
-    new_trk.set_endcap       ( conv_hits_it->Endcap()     );
-    new_trk.set_sector       ( conv_hits_it->Sector()     );
-    new_trk.set_sector_idx   ( conv_hits_it->Sector_idx() );
-    new_trk.set_mode         ( mode );
-    new_trk.set_mode_inv     ( 0 );
-    new_trk.set_rank         ( 0b0100000 );  // Station 1 hit, straightness 0 (see "rank" in AngleCalculation.cc)
-    new_trk.set_winner       ( maxTracks_ - 1 );  // Always set to the last / lowest track
-    new_trk.set_bx           ( bx_ );
-    new_trk.set_first_bx     ( bx_ );
-    new_trk.set_second_bx    ( bx_ );
-    new_trk.set_zone         ( zone );
-    new_trk.set_ph_num       ( conv_hits_it->Zone_hit() );
-    new_trk.set_ph_q         ( 0b010000 );  // Original "quality_code" from PatternRecognition.cc
-    new_trk.set_theta_fp     ( conv_hits_it->Theta_fp() );
-    new_trk.set_theta        ( conv_hits_it->Theta() );
-    new_trk.set_eta          ( conv_hits_it->Eta() );
-    new_trk.set_phi_fp       ( conv_hits_it->Phi_fp() );
-    new_trk.set_phi_loc      ( conv_hits_it->Phi_loc() );
-    new_trk.set_phi_glob     ( conv_hits_it->Phi_glob() );
-    new_trk.set_track_num    ( maxTracks_ - 1 );
-
-    one_hit_trks.push_back( new_trk );
-    if (int(best_tracks.size()) + int(one_hit_trks.size()) >= maxTracks_)
+
+      EMTFTrack new_trk;
+      new_trk.push_Hit ( *conv_hits_it );
+      
+      EMTFPtLUT empty_LUT = {};
```

Comment:
There could be some performance improvement if the initializer_list is not assigned with an = but I'm not sure.